sycl/include/sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp

```diff
@@ -41,22 +82,30 @@ class annotated_ref<T, detail::properties_t<Props...>> {
   using property_list_t = detail::properties_t<Props...>;
 
 private:
-  T *m_Ptr
-#ifdef __SYCL_DEVICE_ONLY__
-      [[__sycl_detail__::add_ir_annotations_member(
-          detail::PropertyMetaInfo<Props>::name...,
-          detail::PropertyMetaInfo<Props>::value...)]]
-#endif
-      ;
+  T *m_Ptr;
 
 public:
   annotated_ref(T *Ptr) : m_Ptr(Ptr) {}
   annotated_ref(const annotated_ref &) = default;
 
-  operator T() const { return *m_Ptr; }
+  operator T() const {
+#ifdef __SYCL_DEVICE_ONLY__
+    return *__builtin_intel_fpga_sycl_ptr_annotation(
```

Comment:
could you please clarify is builtin fpga specific? if not - why is it called so? if yes - is SYCL_DEVICE_ONLY limitation enough?

Reply:
Right, I think this builtin should not be Fpga specific. I'll rename it to  "__builtin_intel_sycl_ptr_annotation"

Reply:
done